modules/client/src/addresslistVoting/internal/client/estimation.ts

```diff
@@ -103,25 +116,33 @@ export class AddresslistVotingClientEstimation extends ClientCore
   /**
    * Estimates the gas fee of executing an AddressList proposal
    *
-   * @param {IExecuteProposalParams} params
+   * @param {string} proposalId
    * @return {*}  {Promise<GasFeeEstimation>}
    * @memberof AddresslistVotingClientEstimation
    */
   public async executeProposal(
-    params: IExecuteProposalParams,
+    proposalId: string,
   ): Promise<GasFeeEstimation> {
     const signer = this.web3.getConnectedSigner();
     if (!signer) {
       throw new NoSignerError();
     } else if (!signer.provider) {
       throw new NoProviderError();
     }
+
+    if (!isProposalId(proposalId)) {
+      throw new InvalidProposalIdError();
+    }
+    const { pluginAddress, id } = decodeProposalId(
```

Comment:
Could we move the `isProposalId()` checks inside of `decodeProposalId()`?

Reply:
decodeProposalId() checks for validity. no need to check it before decoding

Reply:
`decodeProposalId()` checks for validity. no need to check it before decoding

Reply:
If `decodeProposalId()` already checks for validity, then I would simply remove the `isProposalId()` check before the `decodeProposalId()` call

Reply:
done